### PR TITLE
Self-deactivate the plugin after the functionality has been merged to core

### PR DESCRIPTION
**Pull request by azaozz**
This will depend on where the functions are added in core.

If they are in `wp-admin/includes/*`, they are loaded after plugins are loaded, so the plugin cannot check if they exist before loading. In this case the function names from the plugin cannot be reused for core (results in fatal error). Adding the two or three functions to `wp-admin/includes/update.php` seems to make sense. The current code in this PR is for this case.

If they are in `wp-includes/*` the existing method would work. The check whether plugins and themes auto-updating is (globally) enabled may be in `wp-includes/update.php`.

In both of the above cases some updates to the code in `wp-autoupdates.php` would be needed.

### CODE REVIEW

**Comment by azaozz** [comment]
Just a note that this is not ready for merging. The changes will need update and review after the core patch is ready (the new core function's places and names are known).

**Comment by pbiron** [comment]
> Just a note that this is not ready for merging. The changes will need update and review after the core patch is ready (the new core function's places and names are known).

Just marked this as `Draft` just to be safe.

File: wp-autoupdates.php [hunk 1]
```diff
@@ -17,32 +17,22 @@
...
-
-/**
- * Remove auto-updates data on uninstall.
- */
-function wp_autoupdates_activate() {
-	register_uninstall_hook( __FILE__, 'wp_auto_update_uninstall' );
-}
-register_activation_hook( __FILE__, 'wp_autoupdates_activate' );
-
-function wp_auto_update_uninstall() {
```
Comment (audrasjb) [inline wp-autoupdates.php]: @pbiron I don't think we want to remove this part 😉

**Comment by pbiron** [comment]
@azaozz this is now ready to review

**Comment by azaozz** [comment]
Looks good.

**Review by audrasjb — COMMENTED** [review]
@pbiron you removed `wp_auto_update_uninstall` function?

File: wp-autoupdates.php (hunk 1 above)
Comment (pbiron) [reply wp-autoupdates.php]: Git didn't show that to me when it reported merge conflicts.  Let me try this again.

**Review by audrasjb — APPROVED** [review]
Looks good to me 👍

**Pull request merged**